iree/compiler/Dialect/Flow/Transforms/test/dispatch_linalg_on_tensors_dynamic.mlir

```diff
@@ -13,7 +13,7 @@ func @tensor(%arg0 : tensor<?x?xf32>, %arg1 : tensor<?x?xf32>,
 // CHECK-SAME:   %[[ARG1:[a-zA-Z0-9_]+]]: tensor<?x?xf32>
 // CHECK-SAME:   %[[ARG2:[a-zA-Z0-9_]+]]: tensor<?x?xf32>
 //      CHECK:   flow.dispatch.workgroups
-// CHECK-SAME:     (%[[ARG0]], %[[ARG1]], %[[ARG2]])
+// CHECK-SAME:     (%[[ARG2]], %[[ARG1]], %[[ARG0]])
```

Comment:
It's weird these are reversed - is there a reason for that? It looks like the usage is `%arg0, %arg1` in the matmul too, so that they are transposed here is odd. I know it's still correct, but when reading through IR these kind of things add a mental tax :)

Reply:
The original order was because I took the values and explicitly did a reverse of that. I am not sure I remember the rationale for that, but it was mostly to keep it consistent with what `valuesDefinedFromAbove` gives. But that was a hack. There is really no way of enforcing the order. Even `valuesDefinedFromAbove` does not actually specify that there is any ordering there. It was all mostly happy coincidence :) .  I can go back to adding the `reverse` but the reasoning there is shaky.

Reply:
If it's consistently inverting the order I think reversing is fine - it's not a correctness thing, but does make it much easier to read and track the values across the various lowering stages (especially after outlining, where one can't see the usage of the values on the inside anymore). Even if only ordered right 75% of the time that's still nicer than being ordered wrong 100% of the time :)

The improvements here are great so it wouldn't be worth doing too much to fix, but if it's adding a reverse that seems worth it :)